2_task/arrayOperation.js

```diff
@@ -0,0 +1,156 @@
+//function isArray
+function isArrayByInstance(arr) {
+  return arr instanceof Array;
+}
+
+function isArrayByPrototype(arr) {
+  return Object.prototype.toString.call(arr) === '[object Array]';
+}
+
+function isArrayByDuckTyping(arr) {
+  if (arr.splice) {
+    return true;
+  }
+  return false;
+}
+//function range
+function range(start, end, multiply) {
+  function createArray(start, end, multiplay) {
+    let array = [];
+    for (start; start <= end; start += multiplay) {
+      array.push(start);
+    }
+    return array;
+  }
+  let arr = [];
+  multiply = multiply || 1;
+  if (end === undefined) {
+    arr = createArray(0, (start - 1), multiply);
+  } else {
+    arr = createArray(start, end, multiply);
+  }
+  return arr;
+}
+//function compact
+function compactWithoutCycle(array) {
+  return array.filter(function(item) {
+    return item;
+  });
+}
+
+function compactWithCycle(array) {
+  let arrayTrue = [];
+  for (let i = 0; i < array.length; i++) {
+    if (array[i]) {
+      arrayTrue.push(array[i]);
+    }
+  };
+  return arrayTrue;
+}
+//function sum
+function sumWithCycle(array) {
+  let sum = 0;
+  for (let i = 0; i < array.length; i++) {
+    sum += array[i];
+  }
+  return sum;
+}
+
+function sumWithoutCycle(array) {
+  return array.reduce(function(sum, current) {
+    return sum + current;
+  });
+}
+//function unique
+function uniqueWithInnerCycle(array) {
+  let arrayUnique = [];
+  for (let i = 0; i < array.length; i++) {
+    if (arrayUnique.indexOf(array[i]) == -1) {
+      arrayUnique.push(array[i]);
+    }
+  }
+  return arrayUnique;
+}
+
+function uniqueWithMethods(array) {
+  return array.filter(function(value, index, self) {
+    return self.indexOf(value) === index;
+  });
+}
+
+function uniqueWithObject(a) {
+  var prims = {
+      "boolean": {},
+      "number": {},
+      "string": {}
+    },
+    objs = [];
+  return a.filter(function(item) {
+    var type = typeof item;
+    if (type in prims)
+      return prims[type].hasOwnProperty(item) ? false : (prims[type][item] = true);
+    else return objs.indexOf(item) == 0 ? false : objs.push(item);
+  });
+}
+//function last
+function lastWithLength(array) {
+  return array[array.length - 1];
+}
+
+function lastWitSlice(array) {
+  return array.splice(-1).join();
```

Comment:
again, I'm ok to remove this one, it's enough to have `return array[array.length - 1];`